state_machine/state_engine.py

```
@@ -175,15 +184,35 @@ def set_phase(self, phase: MissionPhase) -> Dict[str, Any]:
-            logger.error(f"State transition error: {e.message}")
+            logger.error(
+                f"State transition error: {e.message}",
+                extra={
+                    "component": "state_machine",
+                    "error_type": "transition_invalid",
+                    "previous_phase": previous_phase.value,
+                    "requested_phase": phase.value,
+                    "current_state": self.current_state.value,
+                },
+                exc_info=False,
+            )
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)